src/if.js

```diff
@@ -43,20 +43,42 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {      
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          delete this.__queuedChanges;       
+          return;
+        }        
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();          
```

Comment:
> I suspect, i'm missing smth really obvious

Yeah, concurrency it is ;) 
> 1) let queuedChanges = this.__queuedChanges = []; always assigns empty array to queuedChanges

This is true, but before callback is executed, newValue is pushed to the empty array and therefore when executing `runQueuedChanges` callback, my second assumption doesn't hold.

Thanks for explaining it and sorry for wasting Your time.